crates/analyzer-core/src/base_abstractions.rs

```diff
@@ -30,3 +34,13 @@ pub struct Diagnostic {
 	pub severity: Severity,
 	pub message: String,
 }
+
+/// Represents an included dependency.
+#[derive(Clone, PartialEq, Eq)]
+pub struct IncludedDependency {
+	/// The [`FileId`] of the included target.
+	pub file: FileId,
+
+	/// The path of the included target.
+	pub include_path: String,
```

Comment:
You can only retrieve the path from a `FileId` if you have the `Database` since you'd call `file_id.path(db)` to retrieve it. The `IncludeDependency` is used outside of the `Analyzer` so thought it best to allow access this way.

Is it best to provide a function on `Analyzer` instead? Perhaaps:

```rust
pub fn path_for_file(&self, file_id: FileId) -> String {
  file_id.path(self.db)
}
```

Reply:
There's already a `path(&self, file_id: FileId) -> String` method for `Analyzer`.

Reply:
I've now refactored this. `IncludeDependency` is now a newtype for `FileId` and consuming code uses the `Analyzer::path(...)` to obtain the path.